spec/text-document-spec.coffee

```diff
@@ -2,28 +2,51 @@ Point = require "../src/point"
 TextDocument = require "../src/text-document"
 
 describe "TextDocument", ->
-  document = null
-
-  beforeEach ->
-    document = new TextDocument
+  [document, layer] = []
 
   describe "::buildDisplayLayer()", ->
     describe "::slice(start, end)", ->
       it "returns the content between the start and end points", ->
+        document = new TextDocument(softWrapColumn: 10, tabLength: 4)
         document.setText("""
           abcd\tefg
 
           hij
         """)
-
-        layer = document.buildDisplayLayer(
-          softWrapColumn: 10
-          tabLength: 4
-        )
-
-        expect(layer.getLines()).toEqual [
+        expect(document.displayLayer.getLines()).toEqual [
           "abcd\t   "
           "efg\n"
           "\n"
           "hij"
         ]
+
+  describe "position translation", ->
+    beforeEach ->
+      document = new TextDocument(softWrapColumn: 100, tabLength: 2)
+      document.setText("zero\none\r\ntwo\nthree")
+
+    describe "::characterIndexForPosition(position)", ->
+      it "returns the absolute character offset for the given position", ->
+        expect(document.characterIndexForPosition(Point(0, 0))).toBe 0
```

Comment:
It's definitely what we should have done, which is why I've adopted it here. I don't think it's something we can change before 1.0, but maybe for 2.0. We *can* however update all core code and bundled packages to just use points directly and steer people in that direction. Then someday we can deprecate it.